hardware_interface/src/resource_manager.cpp

```diff
@@ -51,116 +86,492 @@ class ResourceStorage
+  bool initialize_hardware(const HardwareInfo & hardware_info, HardwareT & hardware)
+  {
+    RCUTILS_LOG_INFO_NAMED(
+      "resource_manager", "Initialize hardware '%s' ", hardware_info.name.c_str());
+
+    bool result = hardware.initialize(hardware_info).id() ==
+                  lifecycle_msgs::msg::State::PRIMARY_STATE_UNCONFIGURED;
+
+    if (result)
+    {
+      RCUTILS_LOG_INFO_NAMED(
+        "resource_manager", "Successful initialization of hardware '%s'",
+        hardware_info.name.c_str());
+    }
+    else
+    {
+      RCUTILS_LOG_INFO_NAMED(
+        "resource_manager", "Failed to initialize hardware '%s'", hardware_info.name.c_str());
+    }
+    return result;
+  }
+
+  template <class HardwareT>
+  bool configure_hardware(HardwareT & hardware)
+  {
+    bool result = trigger_hardware_state_transition(
+      std::bind(&HardwareT::configure, &hardware), "configure", hardware.get_name(),
+      lifecycle_msgs::msg::State::PRIMARY_STATE_INACTIVE);
+
+    if (result)
+    {
+      // TODO(destogl): is it better to check here if previous state was unconfigured instead of
+      // checking if each state already exists? Or we should somehow know that transition has
+      // happened and only then trigger this part of the code?
+      // On the other side this part of the code should never be executed in real-time critical
+      // thread, so it could be also OK as it is...
+      // @bmagyar what do you think?
+      for (const auto & interface : hardware_info_map_[hardware.get_name()].state_interfaces)
+      {
+        // add all state interfaces to available list
+        auto found_it = std::find(
+          available_state_interfaces_.begin(), available_state_interfaces_.end(), interface);
+
+        if (found_it == available_state_interfaces_.end())
+        {
+          available_state_interfaces_.emplace_back(interface);
+          // TODO(destogl): change output to debug
+          RCUTILS_LOG_INFO_NAMED(
+            "resource_manager", "(hardware '%s'): '%s' state interface added into available list",
+            hardware.get_name().c_str(), interface.c_str());
+        }
+        else
+        {
+          // TODO(destogl): do here error management if interfaces are only partially added into
+          // "available" list - this should never be the case!
+          RCUTILS_LOG_WARN_NAMED(
+            "resource_manager",
+            "(hardware '%s'): '%s' state interface already in available list."
+            " This can happen due to multiple calls to 'configure'",
+            hardware.get_name().c_str(), interface.c_str());
+        }
+      }
+
+      // add "non-movement" command interfaces to available list
+      for (const auto & interface : hardware_info_map_[hardware.get_name()].command_interfaces)
+      {
+        // check if interface is non-movement interface
+        if (
+          std::find_if(
+            MOVEMENT_INTERFACES.begin(), MOVEMENT_INTERFACES.end(), MOVEMENT_INTERFACES_COMPARE) ==
```

Comment:
I'm really not happy about us creating/maintaining a standard set of movement interfaces that is handled specially.
The default should always be movement interfaces. How hardware components handle/allow for non-movement interfaces should be designed carefully and not in this PR. I'm sorry but I think the best option at this point is to remove everything related to movement interfaces from this PR and we should discuss the design for this instead. This is way too much code IMO to properly assess and maintain every behaviour of this without a design.